cs/src/core/Serializer.cs

```diff
@@ -92,7 +99,52 @@ public Serializer(Type type, bool inlineNested)
         /// </remarks>
         public void Serialize(object obj, W writer)
         {
-            serialize[0](obj, writer);
+            helper.Serialize(obj, writer);
+        }
+
+        SerializerHelper helper;
```

Comment:
Minor: can we put `helper` up at the top of the class where `serialize` used to be.